tests/integration/key_assignment.go

```diff
@@ -166,6 +186,15 @@ func (s *CCVTestSuite) TestKeyAssignment() {
 				if err != nil {
 					return err
 				}
+
+				// check that the key was not assigned to the second validator
+				valConsAddr2, getConsAddrErr := validator2.GetConsAddr() // make sure we don't override err, which we are saving for below
```

Comment:
the expected failure of this test should happen in line #185, so this path should never be executed and the result from that call should be returned and evaluated but the caller, no?
If this is to be considered as a double check beside the check on the error code then `err` should be returned afterwards - this would also make sense to the comment here where in
current implementation it is not overrideing `err` as stated.